modulemd/v2/modulemd-component.c

```diff
@@ -236,6 +275,7 @@ modulemd_component_class_init (ModulemdComponentClass *klass)
   object_class->set_property = modulemd_component_set_property;
 
   klass->copy = modulemd_component_copy_component;
+  klass->equals = modulemd_component_equals;
```

Comment:
This is recursive and definitely not what you want. Look at line 264. I think you meant to have `modulemd_component_default_equals()` here.
```suggestion
  klass->equals = modulemd_component_default_equals;
```